test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1

```
@@ -0,0 +1,400 @@
+
+        $ers | Should Be $null
+        $recipient.Certificates.Count | Should Be 0
+    }
+
+    It "Verify Protect-CmsMessage emits recipient errors" {
+        try {
+            "Hello World" | Protect-CmsMessage -To "SomeThumbprintThatDoesNotExist" -ErrorAction Stop
+            throw "No Exception!"
+        } catch {
+            $_.FullyQualifiedErrorId | Should Be "NoCertificateFound,Microsoft.PowerShell.Commands.ProtectCmsMessageCommand"
+        }
+    }
+
+    It "Verify CmsMessage cmdlets works with paths" {
+        try {
+            $randomNum = Get-Random -Minimum 1000 -Maximum 9999
+            $tempPath = Join-Path $TestDrive "$randomNum-Path-Test-File"
+            $encryptedPath = $tempPath + ".encrypted.txt"
+            "Hello World","How are you?" | Set-Content $tempPath
+
+            Protect-CmsMessage -Path $tempPath -To $certLocation -OutFile $encryptedPath
+
+            $message = Get-CmsMessage -LiteralPath $encryptedPath
+            $message.Recipients[0].IssuerName | Should Be "CN=MyDataEnciphermentCert"
+
+            $decrypted = $message | Unprotect-CmsMessage -To $certLocation
+            $decrypted | Should Be "Hello World`r`nHow are you?`r`n"
+
+            $decrypted = Unprotect-CmsMessage -Path $encryptedPath -To $certLocation
+            $decrypted | Should Be "Hello World`r`nHow are you?`r`n"
+        } finally {
+            Remove-Item $tempPath, $encryptedPath -Force -ErrorAction SilentlyContinue
+        }
+    }
+
+    It "Verify Unprotect-CmsMessage works with local store" {
+        try {
+            $randomNum = Get-Random -Minimum 1000 -Maximum 9999
+            $tempPath = Join-Path $TestDrive "$randomNum-Path-Test-File"
+            "Hello World" | Protect-CmsMessage -To $certLocation -OutFile $tempPath
+
+            # Decrypt using $importedCert in the Cert store
+            $decrypted = Unprotect-CmsMessage -Path $tempPath
+            $decrypted | Should Be "Hello World"
+        } finally {
```

The same about silently skip.